examples/linkaudio/AudioPlatform_Jack.cpp

```diff
@@ -0,0 +1,170 @@
+/* Copyright 2016, Ableton AG, Berlin. All rights reserved.
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *  If you would like to incorporate Link into a proprietary software application,
+ *  please contact <link-devs@ableton.com>.
+ */
+
+#include "AudioPlatform_Jack.hpp"
+#include <chrono>
+#include <iostream>
+#include <string>
+
+namespace ableton
+{
+namespace linkaudio
+{
+
+AudioPlatform::AudioPlatform(Link& link)
+  : mEngine(link)
+  , mSampleTime(0.)
+  , mpJackClient(NULL)
+  , mpJackPorts(NULL)
+{
+  mEngine.setSampleRate(44100.);
+  mEngine.setBufferSize(512);
+  initialize();
+  start();
+}
+
+AudioPlatform::~AudioPlatform()
+{
+  stop();
+  uninitialize();
+}
+
+int AudioPlatform::audioCallback(jack_nframes_t nframes, void* pvUserData)
+{
+  AudioPlatform* pAudioPlatform = static_cast<AudioPlatform*>(pvUserData);
+  return pAudioPlatform->audioCallback(nframes);
+}
+
+int AudioPlatform::audioCallback(jack_nframes_t nframes)
+{
+  using namespace std::chrono;
+  AudioEngine& engine = mEngine;
+
+  const auto hostTime = mHostTimeFilter.sampleTimeToHostTime(mSampleTime);
+
+  mSampleTime += nframes;
+
+  const auto bufferBeginAtOutput = hostTime + engine.mOutputLatency;
```

Comment:
`mOutputLatency` is never set. AFAIK there is no way to query this directly from JACK as it actually depends on the host driver. So it is probably ok to just derive it from the buffer size.